### PR TITLE
Nmea

### PR DESCRIPTION
**Pull request by denysxftr**
Added support for initialization with nmea and dms format.

### CODE REVIEW

File: lib/libgeo/coordinate/class_methods.rb [hunk 1]
```diff
@@ -50,11 +99,31 @@ def min_with_sec(value)
       end
 
       ##
+      # Private: calculate minutes and seconds from nmea coord
```
Comment (Jesterovskiy) [inline lib/libgeo/coordinate/class_methods.rb]: Maybe add Params and Returns?

File: lib/libgeo/coordinate/class_methods.rb [hunk 2]
```diff
@@ -50,11 +99,31 @@ def min_with_sec(value)
...
+
+      ##
       # Private: detect direction based on neg/pos
       #
       def dir_from(degrees)
         degrees < 0 ? :< : :>
       end
+
+      ##
+      # Private: detect direction based on nmea or dms notation
```
Comment (Jesterovskiy) [inline lib/libgeo/coordinate/class_methods.rb]: Same here

**Comment by Jesterovskiy** [comment]
:+1: But have question: if I send wrong coordinates, for example '03920.56074,A' or "5885°39′13.5 A" - where it explode?

File: spec/unit/libgeo/latitude_spec.rb [hunk 3]
```diff
@@ -32,6 +32,94 @@
     end
   end
 
+  describe '.dms' do
```
Comment (Jesterovskiy) [inline spec/unit/libgeo/latitude_spec.rb:35]: Test when send only degrees

**Comment by denysxftr** [comment]
In case of wrong parameters in most of cases it will has wrong coordinates. I added some possible formats of inputs in tests.

Maybe we need a full validation of input string with regexps.

**Comment by Jesterovskiy** [comment]
:+1:

**Pull request merged**